components/org.wso2.carbon.siddhi.editor.core/src/main/resources/web/editor/js/design-view/cron-generator/cron-generator.js

```diff
@@ -0,0 +1,836 @@
+            return list.every(function (month){
+                if(month.includes('/')) {
+                    if(month.endsWith('/')){
+                        errorMsg = "Month values should not end with / ";
+                    } else {
+                        errorMsg = "Month values must be between 1 and 12";
+                    }
+                    var startingDayOfMonthOptionArr = month.split('/');
+                    var isValidElements = (validateRangeOfDay([startingDayOfMonthOptionArr[0]], 1, 12) &&
+                                                        validateRangeOfDay([startingDayOfMonthOptionArr[1]], 1, 12));
+                    var isValidFirstElem = (startingDayOfMonthOptionArr[0] === '*' &&
+                                                        validateRangeOfDay([startingDayOfMonthOptionArr[1]], 1, 12));
+                    return isValidElements || isValidFirstElem;
+                } else if(month.includes('-')) {
+                    if(month.endsWith('-')){
+                        errorMsg = "Month values should not end with - ";
+                    } else {
+                        errorMsg = "Month values must be between 1 and 12 or Jan - Dec";
+                    }
+                    var monthRangeArr = month.split('-');
+                    var validMonthRange = parseInt(monthRangeArr[0]) < parseInt(monthRangeArr[1]);
+                    var validMonthStrRange = monthArray.indexOf(monthRangeArr[0]) < monthArray.indexOf(monthRangeArr[1]);
+
+                    return !isNaN(parseInt(monthRangeArr[0])) && !isNaN(parseInt(monthRangeArr[1])) ?
+                        validateRangeOfDay(monthRangeArr, 1, 12) && validMonthRange :
+                        validateStringValue(monthRangeArr, monthArray) && validMonthStrRange;
+                } else {
+                    if(month === ""){
+                        errorMsg = "Month values should not end with , ";
+                    } else {
+                        errorMsg = "Month values must be between 1 and 12 or Jan - Dec";
+                    }
+                    var firstIndexValue = month.charAt(0);
+                    var secondIndexValue = month.charAt(1);
+                    if(month.length === 1){
+                        return validateRangeOfDay([month], 1, 12) || month === '*';
+                    } else {
+                        return (!isNaN(parseInt(firstIndexValue)) && !isNaN(parseInt(secondIndexValue))) ?
+                                validateRangeOfDay([month], 1, 12) : validateStringValue([month], monthArray);
+                    }
+                }
+            });
+        };
+
+        var weekArray = ['mon', 'tue', 'wed', 'thu', 'fri', 'sat','sun'];
+
+        var validateDayOfWeekValue = function(dayOfWeek, dayOfMonth) {
+            if(dayOfMonth === '?' && dayOfWeek.includes('L')) {
+                if(dayOfWeek.includes(',') && dayOfMonth === '?'){
+                    errorMsg = "Invalid cron expression";
```

Comment:
Use same error message standard and explain why its invalid as i mentioned above

Reply:
fixed in e096cb9